src/main/java/com/marklogic/kafka/connect/source/RowBatcherSourceTask.java

```diff
@@ -65,7 +77,7 @@ public List<SourceRecord> poll() throws InterruptedException {
 
         long start = System.currentTimeMillis();
         try {
-            rowBatcher = newRowBatcher(newSourceRecords);
+            rowBatcher = newRowBatcher(previousBatchMaxValue, newSourceRecords);
```

Comment:
I think newSourceRecords should be first since it will never be null, while the previous value may be null. I generally prefer ordering params based on required stuff first, optional stuff second.